### PR TITLE
fix: binary charset [SUP-5410]

### PR DESCRIPTION
**Pull request by sigalikanevsky**

### CODE REVIEW

**Review by eitamring — COMMENTED** [review]
Good call on detecting invalid UTF-8 to prevent decoder panics.
Covers common UTF-8 family variants (utf8, utf8mb3, utf8mb4)
Clean isolation of fallback logic in its own function

Fix fallback: Use either charmap.Windows1252 or lossless hex escape (\xNN) for invalid UTF-8.

Avoid double decode: Pass decoded raw instead of re-slicing.
Name/style polish: Rename helpers for clarity.

**Review by eitamring — APPROVED** [review]

**Review by aaronabv — COMMENTED** [review]
Delete please the `.!93441!main.go` file

File: replication/row_event_test.go [hunk 1]
```diff
@@ -1613,3 +1613,49 @@ func TestDecodeByCharSet(t *testing.T) {
...
+		t.Run(tc.name, func(t *testing.T) {
+			got := bytesToLatin1String(tc.in)
+			if got != tc.want {
+				t.Fatalf("bytesToLatin1String() = %q, want %q", got, tc.want)
+			}
+		})
+	}
+}
+
+func TestDecodeValueBinaryLatin1Fallback(t *testing.T) {
```
Comment (aaronabv) [inline replication/row_event_test.go:1638]: extend the test not to include only latin1 please

**Review by eitamring — APPROVED** [review]

**Comment by OhadPerryBoomi** [comment]
## Complexity Analysis

**Score:** 5/10

Modifies charset handling logic across multiple Go files with non-trivial SQL query changes (adding JOINs and CASE logic for binary types), introduces a new sanitization function for non-printable characters, fixes a datetime parsing bug, and adds comprehensive test coverage; moderate complexity due to cross-cutting charset/encoding concerns and edge case handling.
